crypto/src/lib.rs

```diff
@@ -918,15 +925,19 @@ mod tests {
             TestJson {
                 public_key: PublicKey {
                     digest_function: Algorithm::BlsSmall,
-                    payload: hex_decode(
-                        "040CB3231F601E7245A6EC9A647B450936F707CA7DC347ED258586C1924941D8BC38576473A8BA3BB2C37E3E121130AB67103498A96D0D27003E3AD960493DA79209CF024E2AA2AE961300976AEEE599A31A5E1B683EAA1BCFFC47B09757D20F21123C594CF0EE0BAF5E1BDD272346B7DC98A8F12C481A6B28174076A352DA8EAE881B90911013369D7FA960716A5ABC5314307463FA2285A5BF2A5B5C6220D68C2D34101A91DBFC531C5B9BBFB2245CCC0C50051F79FC6714D16907B1FC40E0C0"
+                    payload: ImmutableBytes::new(
+                        hex_decode(
+                            "040CB3231F601E7245A6EC9A647B450936F707CA7DC347ED258586C1924941D8BC38576473A8BA3BB2C37E3E121130AB67103498A96D0D27003E3AD960493DA79209CF024E2AA2AE961300976AEEE599A31A5E1B683EAA1BCFFC47B09757D20F21123C594CF0EE0BAF5E1BDD272346B7DC98A8F12C481A6B28174076A352DA8EAE881B90911013369D7FA960716A5ABC5314307463FA2285A5BF2A5B5C6220D68C2D34101A91DBFC531C5B9BBFB2245CCC0C50051F79FC6714D16907B1FC40E0C0"
+                        )
+                        .expect("Failed to decode public key.")
                     )
-                    .expect("Failed to decode public key.")
                 },
                 private_key: PrivateKey {
                     digest_function: Algorithm::BlsSmall,
-                    payload: hex_decode("0000000000000000000000000000000060F3C1AC9ADDBBED8DB83BC1B2EF22139FB049EECB723A557A41CA1A4B1FED63")
-                    .expect("Failed to decode private key"),
+                    payload: ImmutableBytes::new(
```

Comment:
This pattern is repeated in tests a lot:

```rs
ImmutableBytes::new(hex_decode("deadbeef").expect("failed"));
```

Maybe, define a test-only method for it?

```rs
impl ImmutableBytes {
    #[cfg(test)]
    fn from_hex(hex: AsRef<str>) -> Self {
        ...
    }
}
```

Reply:
Made it into a regular fallible method, as it had a use outside of tests